scripts/configurator/parser/cmake_var.py

```diff
@@ -1,9 +1,92 @@
-from typing import Dict, Any, Type, Mapping
+from typing import Dict, Any, Optional, Type, Mapping, List, Callable
+import re
+from sys import stderr
+
+
+class metadata:
+   keyword_expr = re.compile(r"^([A-Z_]+) (.*)")
+
+   def __init__(self, comment: List[str]) -> None:
+      self.depends_on: Optional[List[str]] = None
+      self.group: Optional[str] = None
+      self.realtype: Optional[str] = None
+
+      self.kw_to_setter: Mapping[str, Callable]= {
+         "REALTYPE": self.get_setter_realtype(),
+         "DEPENDS_ON": self.get_setter_depends_on(),
+         "GROUP": self.get_setter_group(),
+      }
+
+      self.kw_to_var: Mapping[str, Any] = {
+         "REALTYPE": self.realtype,
+         "DEPENDS_ON": self.depends_on,
+         "GROUP": self.group,
+      }
+
+      for i, line in enumerate(comment):
+         match = self.keyword_expr.match(line)
+         if not match:
+            continue
+
+         key = match.group(1)
+         if i == 0:
+            if key in self.kw_to_setter:
+               msg = """WARNING: First line uses a reserved keyword: {}.
+               This will be ignored."""
+               stderr.write(msg.format(key))
+            continue
+
+         elif key not in self.kw_to_setter:
+            msg = "{} is not a valid metadata keyword."
+            raise ValueError(msg.format(key))
+
+         elif self.kw_to_var[key] is not None:
+            msg = "Key {} has already a value associated to itself."
+            raise ValueError(msg.format(key))
+
+         setter = self.kw_to_setter[key]
+         s = match.group(2)
+         if key == "DEPENDS_ON":
```

Comment:
What's this? You should use the parser in the kw dict.
Like `p = kw[key].parser`  and later `setter(p(s))` or something like that.